addons/sale_stock/models/sale_order_line.py

```diff
@@ -264,7 +264,7 @@ def _prepare_procurement_values(self, group_id=False):
             'route_ids': self.route_id,
             'warehouse_id': self.order_id.warehouse_id or False,
             'partner_id': self.order_id.partner_shipping_id.id,
-            'product_description_variants': self.with_context(lang=self.order_id.partner_id.lang)._get_sale_order_line_multiline_description_variants(),
+            'product_description_variants': self.name if self.name != self.product_id.name else '',
```

Comment:
This check seems way to weaker. There is a lot of cases where self.product_id.name doesn't match the line name. The base definition of line is already:
        name = self.display_name
        if self.description_sale:
            name += '\n' + self.description_sale
So you should already use the display_name instead of name and if a description sale is present, it would returns too much useless data

Reply:
+ All the options and ...

Reply:
Thanks for the review @amoyaux !

You are totally right ! In addition, you made me realize that the name of the SOL is computed regarding the language of the `line.order_id.partner_id`: 
https://github.com/odoo/odoo/blob/2d6e7c75eeb87cd85fa28ca0519a9734554c3437/addons/sale_management/models/sale_order_line.py#L21-L22
while the name and display name of the product depends on the language of the context of the product itself. Thus, it is also important to add a contextual language in the comparison !